src/main/java/org/apache/commons/imaging/ImagingParameters.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.imaging;
+
+import org.apache.commons.imaging.common.BufferedImageFactory;
+
+/**
+ * Imaging parameters.
+ *
+ * <p>Contains parameters that are common to all formats. Implementations must include
+ * the specific parameters for each image format.</p>
+ *
+ * @since 1.0-alpha3
+ */
+public class ImagingParameters {
+
+    /**
+     * Whether to throw an exception when any issue occurs during reading
+     * or writing a file format. Default is {@code false}.
+     */
+    private boolean strict = false;
+
+    /**
+     * An optional file name, used for the description of input streams
+     * where a file name would be hard (or not possible) to be identified.
+     * Default is {@code null}.
+     */
+    private String fileName = null;
+
+    /**
+     * Factory to create {@code BufferedImage}s. Default is {@code null}.
+     */
+    private BufferedImageFactory bufferedImageFactory = null;
+
+    /**
+     * Image format used in write operations to indicate desired image format.
+     * Default is {@code null}.
+     *
+     * <p>Valid values: Any format defined in ImageFormat, such as
+     * ImageFormat.IMAGE_FORMAT_PNG.</p>
+     *
+     * @see org.apache.commons.imaging.ImageFormats
+     */
+    private ImageFormat imageFormat;
```

Comment:
Should this be final? I'm wondering since otherwise it would be possible to create an instance with a class name that doesn't match the image format type. Ex:
```java
PngImagingParameters params = new PngImagingParameters();
params.setImageFormat(ImageFormat.BMP); // doesn't seem like a good thing
```

Perhaps something like this would work?
```java
public abstract class ImagingParameters {
    private final ImageFormat imageFormat;

    protected ImagingParameters(final ImageFormat imageFormat) {
        this.imageFormat = imageFormat;
    }
}

public class PngImageParameters {
    public PngImageParameters() {
        super(ImageFormat.PNG);
    }
}
```

Reply:
Excellent idea! That reduced the code smell, where I had a `setImageFormat` in the constructor of each parameter class and, as you pointed, users could create objects with invalid states. Thanks @darkma773r!